Common subdirectories: /data/foundry/samsung/sf5a/ip/synopsys/dwc_pcie4phy_sssf5a_x4ns/Latest/upcs/sim/cust_tb_regr and /scratch/workspace/ferguson/dwc_ap_lpddr5x_phy_sssf5a/upcs/sim/cust_tb_regr
diff -uN /data/foundry/samsung/sf5a/ip/synopsys/dwc_pcie4phy_sssf5a_x4ns/Latest/upcs/sim/run_ate_cfg_ref_100m.sh /scratch/workspace/ferguson/dwc_ap_lpddr5x_phy_sssf5a/upcs/sim/run_ate_cfg_ref_100m.sh
--- /data/foundry/samsung/sf5a/ip/synopsys/dwc_pcie4phy_sssf5a_x4ns/Latest/upcs/sim/run_ate_cfg_ref_100m.sh	2024-09-04 16:09:00.281208434 +0000
+++ /scratch/workspace/ferguson/dwc_ap_lpddr5x_phy_sssf5a/upcs/sim/run_ate_cfg_ref_100m.sh	2024-11-18 12:33:52.789742035 +0000
@@ -13,14 +13,29 @@
 
 ATE_TEST_LIST="ate_b2b_test"
 # Tests without PHY powered-up
+#ATE_PHY_CFG+="\n
+#<TEST: JTAG_REG_NOCLK_CLK>                                                          \
+#ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ LOAD_FW             + \
+#ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ JTAG_NOCLK          + \
+#ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ JTAG_CLK            + \
+#ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ REG                 + \
+#"
+#ATE_PHY_CFG+="\n
+#<TEST: JTAG_REG_NOCLK>                                                          \
+#ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ LOAD_FW             + \
+#ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ JTAG_NOCLK          + \
+#ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ REG                 + \
+#"
+
 ATE_PHY_CFG+="\n
-<TEST: JTAG_REG>                                                                    \
+<TEST: JTAG_REG_CLK>                                                          \
 ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ LOAD_FW             + \
-ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ JTAG_NOCLK          + \
 ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ JTAG_CLK            + \
 ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ REG                 + \
 "
 
+
+
 # Analog tests with PHY powered-up
 ATE_PHY_CFG+="\n
 <TEST: PCIE_ELB_RXDET_SIGDET_DEEMPH>                                                \
@@ -30,6 +45,24 @@
 ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ LFPS_SIGDET         + \
 ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ TXDEEMPH            + \
 "
+ATE_PHY_CFG+="\n
+<TEST: PCIE_ELB_RXDET>                                                \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ INIT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ BERT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ TX_RXDET            + \
+"
+ATE_PHY_CFG+="\n
+<TEST: PCIE_ELB_SIGDET>                                                \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ INIT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ BERT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ LFPS_SIGDET         + \
+"
+ATE_PHY_CFG+="\n
+<TEST: PCIE_ELB_DEEMPH>                                                \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ INIT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ BERT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ TXDEEMPH            + \
+"
 
 ATE_PHY_CFG+="\n
 <TEST: PCIE_ELB_COMPRD_RXMAN_TXMAN_ANAOBSV>                                         \
@@ -42,6 +75,35 @@
 "
 
 ATE_PHY_CFG+="\n
+<TEST: PCIE_ELB_COMPRD>                                         \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ INIT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ BERT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ COMP_READ           + \
+"
+
+ATE_PHY_CFG+="\n
+<TEST: PCIE_ELB_RXMAN>                                         \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ INIT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ BERT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ RX_MAN              + \
+"
+
+ATE_PHY_CFG+="\n
+<TEST: PCIE_ELB_TXMAN>                                                              \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ INIT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ BERT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ TX_MAN              + \
+"
+
+ATE_PHY_CFG+="\n
+<TEST: PCIE_ELB_ANAOBSV>                                         \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ INIT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ BERT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ELB_STD    @ LFSR15  @ ANA_OBSV            + \
+"
+
+
+ATE_PHY_CFG+="\n
 <TEST: PCIE_ILB_CHKPHY_VCM_DCLEV_TXSWING>                                           \
 ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ INIT                + \
 ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ BERT                + \
@@ -50,6 +112,38 @@
 ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ TX_DCLEV            + \
 ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ TX_SWING            + \
 "
+
+
+ATE_PHY_CFG+="\n
+<TEST: PCIE_ILB_CHKPHYG>                                           \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ INIT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ BERT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ CHK_PHY             + \
+"
+
+ATE_PHY_CFG+="\n
+<TEST: PCIE_ILB_VCM>                                           \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ INIT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ BERT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ TX_VCM              + \
+"
+
+
+ATE_PHY_CFG+="\n
+<TEST: PCIE_ILB_DCLEV>                                           \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ INIT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ BERT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ TX_DCLEV            + \
+"
+
+
+ATE_PHY_CFG+="\n
+<TEST: PCIE_ILB_TXSWING>                                           \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ INIT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ BERT                + \
+ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15  @ TX_SWING            + \
+"
+
 ATE_PHY_CFG+="\n
 <TEST: PCIE_RINGO>                                                                  \
 ATE_PCIECM_G1_20B      @ REFCLK_100M @ ILB_INTRA  @ LFSR15         @ INIT         + \
@@ -123,6 +217,24 @@
                  ATE_PCIECM_G4_16B         \
                 ")
 
+
+ref100_testname+=(PCIE_G1)
+ref100_testrate+=("                        \
+                 ATE_PCIECM_G1_20B         \
+                ")
+ref100_testname+=(PCIE_G2)
+ref100_testrate+=("                        \
+                 ATE_PCIECM_G2_20B         \
+                ")
+ref100_testname+=(PCIE_G3)
+ref100_testrate+=("                        \
+                 ATE_PCIECM_G3_16B         \
+                ")
+ref100_testname+=(PCIE_G4)
+ref100_testrate+=("                        \
+                 ATE_PCIECM_G4_16B         \
+                ")
+
 for idx in ${!ref100_testname[@]}; do
 ATE_PHY_CFG+="\n
 "
diff -uN /data/foundry/samsung/sf5a/ip/synopsys/dwc_pcie4phy_sssf5a_x4ns/Latest/upcs/sim/run_ate_regression.sh /scratch/workspace/ferguson/dwc_ap_lpddr5x_phy_sssf5a/upcs/sim/run_ate_regression.sh
--- /data/foundry/samsung/sf5a/ip/synopsys/dwc_pcie4phy_sssf5a_x4ns/Latest/upcs/sim/run_ate_regression.sh	2024-09-04 16:09:26.816113969 +0000
+++ /scratch/workspace/ferguson/dwc_ap_lpddr5x_phy_sssf5a/upcs/sim/run_ate_regression.sh	2024-10-15 18:59:20.205125878 +0000
@@ -30,7 +30,7 @@
    echo "    -sim_args       <sim_args>     (default '')"
    echo " "
    echo "Example command:"
-   echo "    run_ate_regression.sh -proto mp_upcs -cfgfile run_ate_cfg_ref_24m -sim_type gtech -sim_tool vcs -acc_type 0 -fw_sim_mode fast -grid 1"
+   echo "    run_ate_regression.sh -proto mp_phy -cfgfile run_ate_cfg_ref_24m -sim_type gtech -sim_tool vcs -acc_type 0 -fw_sim_mode fast -grid 1"
    echo " "
    exit 0
 }
@@ -316,7 +316,8 @@
     if [ $sim_tool == "ncv" ] ; then
         grd_cmd="bsub $queue $cds_lic_opt -o output.o%J"
     else
-        grd_cmd="bsub $queue -R rusage[scratch_free=10] -o output.o%J"
+        #grd_cmd="bsub $queue -R rusage[scratch_free=10] -o output.o%J"
+        grd_cmd="sbatch --export=ALL "
     fi
 fi
 
@@ -559,7 +560,8 @@
     ## Create the array run script
     echo "#!/bin/bash" > $array_script
     echo "# Regression script for ATE tests" >> $array_script
-    echo "regress\$SGE_TASK_ID.sh" >> $array_script
+   # echo "regress\$SGE_TASK_ID.sh" >> $array_script
+    echo "./regress\$SLURM_ARRAY_TASK_ID.sh" >> $array_script
     #
     ## Run the array job
     echo "Running ATE Regression as Grid Array job"
@@ -574,12 +576,14 @@
     ## Create the array run script
     echo "#!/bin/bash" > $array_script
     echo "# Regression script for ATE tests" >> $array_script
-    echo "regress\${LSB_JOBINDEX}.sh" >> $array_script
+    echo "./regress\$SLURM_ARRAY_TASK_ID.sh" >> $array_script
     #
     ## Run the array job
     echo "Running ATE Regression as Grid Array job"
-    echo "$grd_cmd -J ate_array -J ate_array[1-$num_jobs] $array_script"
-    job_id=`$grd_cmd -J ate_array -J ate_array[1-$num_jobs] $array_script  | perl -lane '$F[2] =~ s#(\d+)\..*#$1# ; print $F[1]'`
+    #echo "$grd_cmd -J ate_array -J ate_array[1-$num_jobs] $array_script"
+    #job_id=`$grd_cmd -J ate_array -J ate_array[1-$num_jobs] $array_script  | perl -lane '$F[2] =~ s#(\d+)\..*#$1# ; print $F[1]'`
+    echo "$grd_cmd -J ate_array --array 1-$num_jobs $array_script"
+    job_id=`$grd_cmd -J ate_array --array 1-$num_jobs $array_script | awk '{print $4}'`
     echo ""
     echo " job_id=$job_id " | tee -a $PCS_SCRIPT_PATH/$ATE_TEST_DIR/$regress_name.log
 fi
diff -uN /data/foundry/samsung/sf5a/ip/synopsys/dwc_pcie4phy_sssf5a_x4ns/Latest/upcs/sim/run_ate_test.sh /scratch/workspace/ferguson/dwc_ap_lpddr5x_phy_sssf5a/upcs/sim/run_ate_test.sh
--- /data/foundry/samsung/sf5a/ip/synopsys/dwc_pcie4phy_sssf5a_x4ns/Latest/upcs/sim/run_ate_test.sh	2024-09-04 16:09:26.818113962 +0000
+++ /scratch/workspace/ferguson/dwc_ap_lpddr5x_phy_sssf5a/upcs/sim/run_ate_test.sh	2024-10-16 09:27:06.615833818 +0000
@@ -12,7 +12,7 @@
    echo "                       Note: for 'mp_rmmi' \$RMMI_ROOT \$LDO_ROOT and \$MPHY_MODULE should be set if default value is not suitable"
    echo "         <test_num>    is back-to-back tests number"
    echo "         <cfg_str>     is configuration string with format <TEST?@PROT@RATE@FREQ@LPBK@PATN@TCAS>"
-   echo "         <cfgfile>     is the run_ate_cfg_*.sh file which contain the testcase and test config info"
+   echo "         <cfgfile>     is the ate_cfg_*.sh file which contain the testcase and test config info"
    echo "  Optional arguments:"
    echo "    -model_path <model_path>     (the root path of the package. the setting is necessary when"
    echo "                                  the regression isn't kicked-off from phy/sim or upcs/sim directory)"
@@ -269,7 +269,7 @@
   vcs_run="vcs -full64 +v2k -sverilog +neg_tchk -debug_pp -lca +lint=TFIPC-L -l $log_file -R +rad -Mdir=$vcs_tmp/csrc -o $vcs_tmp/simv"
   ncv_run="ncverilog -ALLOWREDEFINITION +xcelicq -sv -q -l $log_file -access +rwc"
 else
-  vcs_run="vcs -full64 +v2k -sverilog -lca +lint=TFIPC-L -l $log_file -R +rad -Mdir=$vcs_tmp/csrc -o $vcs_tmp/simv"
+  vcs_run="vcs  -debug_access+nomemcbk+dmptf -debug_region+cell -full64 +v2k -sverilog -lca +lint=TFIPC-L -l $log_file -R +rad -Mdir=$vcs_tmp/csrc -o $vcs_tmp/simv"
   ncv_run="ncverilog +xcelicq -sv -q -l $log_file -access +rwc -ALLOWREDEFINITION +nc64bit "
 fi
 
@@ -404,18 +404,18 @@
 fi
 
 # Set up verdi lib
-if [[ $fsdb == 1 ]]; then
-    VERDI=/global/apps
-    echo " "
-    echo "Using FSDB for dumping the file..."
-    VERDI_VERSION=`which verdi | cut -d "/" -f4`
-    echo "Verdi tool version : $VERDI_VERSION"
-    export VERDI_HOME=$VERDI/$VERDI_VERSION
-    echo "Path for Verdi Tool : $VERDI_HOME"
-    echo " "
-    fsdb="1";
-fi
-
+#if [[ $fsdb == 1 ]]; then
+#    VERDI=/global/apps
+#    echo " "
+#    echo "Using FSDB for dumping the file..."
+#    VERDI_VERSION=`which verdi | cut -d "/" -f4`
+#    echo "Verdi tool version : $VERDI_VERSION"
+#    export VERDI_HOME=$VERDI/$VERDI_VERSION
+#    echo "Path for Verdi Tool : $VERDI_HOME"
+#    echo " "
+#    fsdb="1";
+#fi
+#
 # Set up wave capture
 if [[ $waves == 1 ]] ; then
    if [[ $sim_tool == "vcs" ]] ; then
